lib/chef/knife/mixin/helper.rb

```diff
@@ -43,6 +44,35 @@ def values_from_json(json)
       rescue JSON::ParserError
         raise JSON::ParserError, "#{json} is not valid JSON!"
       end
+
+      # I/P: json string
+      # It checks wheather it contains any non-printable character present or not,
+      # If present then raises error of InvalidValue
+      def validate_json(json)
+        begin
+          evaled_json = eval(json)
+        rescue StandardError => e
+          puts e.message
+        end
```

Comment:
What does calling `eval` on the json do, and what would cause a failure that is not fatal?

Reply:
Hi @phiggins, as our vault json contents string with an hash format, using eval converting to the hash.

Reply:
Is there a scenario where `eval` can fail and `evaled_json` is still a valid value?

Reply:
As in vault we accept only hash format, there is less possibility of failing the eval method.